### DISCUSSION TITLE
Questions regarding the 15000 file limit

### DISCUSSION THREAD

**Discussion by ScottAwesome** [vercel/community#1083] [Help]
Hello!

For context: We are looking very closely at adopting Next.js & Vercel as a platform of choice, however we are working a monorepo environment, and we keep all of our UI / UX code in one monorepo, along with configurations helper packages etc. 
We use `nx` to cache builds and we like that Vercel allows us to integrate our build tools and can easily scale out our serverless components as well (via API routes)

As I was reading about the Vercel platform, the following caught my eye. 

According to [Vercel Docs](https://vercel.com/docs/concepts/limits/overview#files) if you are deploying via the Vercel CLI, there is a 15,000 file limit on source files for creating a deployment.

The wording here is odd to me on this page, as it doesn't actually define what a *source file* is in this context, and I have the following questions:

1. does this mean only *deployment assets*, e.g., post-build files that _actually_ deployed? or is this all source code in a repository? 15K is actually alot easier to hit than you might think here.

2. if you're using the GitHub integration, are you subject to the same limitations?

3. Is there any way around this as a limitation?

I'm not clear if a [previous discussion](https://github.com/vercel/vercel/discussions/8643) is still relevant. It seems semi-related, not sure.

Thanks!

**Comment by leerob** [comment]
Hey! Can you share a bit more about your monorepo setup?

- How many source files are you looking to host?
- How many applications are in your monorepo?
- What domains / sites is this for? How are they stitched together? (Just curious)

**Reply by ScottAwesome** [reply]
@leerob 

Clarification point still: _define source files to host_. Are you talking about production or the overall numer of files in our repo?

If you're talking about the repository, I think we sit pretty close to ~8,000 but we're about to merge another repository into the monorepo and that will definitely push up. We're getting close to the stated limit regardless.

Depends on how you define application, since we split everything up by logical domains (think Domain Driven Design, not URL domains). I'd say if we just map domains to apps we're talking about ~30 apps.

The goal was to use multi-zone deployments, since we can cleanly (for the most part) separate functionality by URL, that's how we would plan to stitch things together, as it would map well to how we already do it.

I'd rather not reveal the _exact_ URLs here in a public forum though.

**Reply by leerob** [reply]
No worries, I understand - you can always email me at lee@vercel.com as well.

**Reply by ScottAwesome** [reply]
@leerob I emailed you.

**Reply by Dvorson** [reply]
Hello!

We're hitting this limit also. Is there any solution?

**Reply by bambrose24** [reply]
I'm also hitting this problem as well, and I don't understand what the problem is. The GitHub integration is able to clone and build/deploy our app(s) just fine, but in trying to set up GitHub Actions to manage our deployments we are hitting this error as it's using the Vercel CLI. Please advise about whether this is something we can expect to change, and if so, when, or any workarounds.

**Reply by nicosilvabuydepa** [reply]
same error here! any suggestion @leerob ?

**Reply by eddiejaoude** [reply]
We just hit the same error on the open source repo LinkFree, here is the [GitHub Action log](https://github.com/EddieHubCommunity/LinkFree/actions/runs/5340949456/jobs/9681290676).  There is some debugging in it also already, but we are keen to understand why and also how to debug further so that we can optimise and continue to deploy our app again

**Reply by valderane** [reply]
Hello, I have the same issue. Did anyone manage to solve it please ? Or at least know where the issue really come from ?

**Reply by rowin1125** [reply]
@valderane if you using the vercel CLI to deploy you can deploy with the following options to solve this issue:

```
vercel deploy --token=${{secrets.VERCEL_TOKEN}} --archive=tgz --prebuilt
```

**Comment by aboutmydreams** [comment]
if use `--archive=tgz` in a big project maybe another problem will happen:

File size limit exceeded (100 MB)
- https://github.com/orgs/vercel/discussions/174
